app/controllers/registrations_controller.rb

```diff
@@ -558,9 +558,24 @@ def stripe_webhook
   end
 
   def payment_completion
-    registration = Registration.includes(:competition).find(params[:id])
+    uses_v2 = params[:attendee_id].present?
```

Comment:
Hm... It seems like you're "overloading" the route params. This existing `payment_completion` route definitely requires a registration ID to be set as part of the route, and it cannot be skipped because it's a hard part of the route structure: `/registrations/:id/payment-completion` (or something like that).
So in the new registration frontend, are you gonna fill a fantasy value for `:id` in the URL? Doesn't feel right.

Instead, may I suggest just changing the route altogether so that it does not include the reg ID at all anymore?
You can use `params[:payment_intent]` which will always be provided as `my/cool/url?payment_intent=123` as GET param by Stripe.
Then, you retrieve the matching Stripe Record from our DB (I think we already do that anyways) and then the `payment_intent` of that Stripe Record.
Payment Intents on the WCA side always have a "holder" which is polymorphic, so you can do a type check on the holder (or simply run string comparisons on the `holder_type` column, which feels even more natural) to find out whether it's a V2 comp being redirected

Reply:
Compare also the Stripe Webhook endpoint, which does not contain any WCA entity's ID in its params and only ever loads the registration based on our PI's holder.